appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/CLIBootstrap.java

```diff
@@ -535,6 +541,27 @@ StringBuilder format(final StringBuilder commandLine, final boolean useQuotes, f
         }
     }
 
+    private class CommandLineArgument extends CommandLineElement {
+        CommandLineArgument(String patternString, int flags) {
+            super(patternString, flags);
+        }
+        @Override
+        StringBuilder format(final StringBuilder commandLine,
+                final boolean useQuotes, final String v) {
+            if (commandLine.length() > 0) {
+                commandLine.append(' ');
+            }
+            commandLine.append((useQuotes ? quoteCommandLineArgument(v) : v));
+            return commandLine;
+        }
+        private String quoteCommandLineArgument(String s) {
```

Comment:
I'd rather see this implemented as top-level static, package-accessible method and have it separately tested (and used here).